src/nvim/screen.c

```diff
@@ -2185,16 +2186,18 @@ static int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow,
       }
     }
 
+    line = ml_get_buf(buf, lnum, false);
     has_decor = decor_redraw_line(wp->w_buffer, lnum-1,
                                   &decor_state);
 
     for (size_t k = 0; k < kv_size(*providers); k++) {
       DecorProvider *p = kv_A(*providers, k);
       if (p && p->redraw_line != LUA_NOREF) {
-        FIXED_TEMP_ARRAY(args, 3);
+        FIXED_TEMP_ARRAY(args, 4);
         args.items[0] = WINDOW_OBJ(wp->handle);
         args.items[1] = BUFFER_OBJ(buf->handle);
         args.items[2] = INTEGER_OBJ(lnum-1);
+        args.items[3] = INTEGER_OBJ(win_linetabsize(wp, line, wp->w_leftcol + wp->w_width - win_col_off(wp)));
```

Comment:
is this really correct? what happens with `set linebreak` ?

Reply:
The CI seems to indicate that it works, is it well tested already ?

Reply:
don't expect CI to contain broken lines, excessive unicode, etc in places where it didn't matter yet :D

Reply:
Should work properly now